python/packages/autogen-ext/src/autogen_ext/agents/azure/_azure_ai_agent.py

```diff
@@ -0,0 +1,986 @@
+
+                async with AIProjectClient.from_connection_string(
+                    credential=credential, conn_str=os.getenv("AI_PROJECT_CONNECTION_STRING", "")
+                ) as project_client:
+                    conn = await project_client.connections.get(connection_name=os.getenv("BING_CONNECTION_NAME", None))
+
+                    bing_tool = models.BingGroundingTool(conn.id)
+                    agent_with_bing_grounding = AzureAIAgent(
+                        name="bing_agent",
+                        description="An AI assistant with Bing grounding",
+                        project_client=project_client,
+                        deployment_name="gpt-4o",
+                        instructions="You are a helpful assistant.",
+                        tools=bing_tool.definitions,
+                        metadata={"source": "AzureAIAgent"},
+                    )
+
+                    result = await agent_with_bing_grounding.on_messages(
+                        messages=[TextMessage(content="What is Microsoft's annual leave policy?", source="user")],
+                        cancellation_token=CancellationToken(),
+                        message_limit=5,
+                    )
+                    print(result)
+
+
+            if __name__ == "__main__":
+                dotenv.load_dotenv()
+                asyncio.run(bing_example())
+
+        Use the AzureAIAgent to create an agent with file search capability:
+
+        .. code-block:: python
+
+            import asyncio
+            import os
+
+            import dotenv
+            from autogen_agentchat.messages import TextMessage
+            from autogen_core import CancellationToken
+            from autogen_ext.agents.azure._azure_ai_agent import AzureAIAgent
+            from azure.ai.projects.aio import AIProjectClient
+            from azure.identity.aio import DefaultAzureCredential
+
+
+            async def file_search_example():
+                credential = DefaultAzureCredential()
+                async with AIProjectClient.from_connection_string(
+                    credential=credential, conn_str=os.getenv("AI_PROJECT_CONNECTION_STRING", "")
+                ) as project_client:
+                    agent_with_file_search = AzureAIAgent(
+                        name="file_search_agent",
+                        description="An AI assistant with file search capabilities",
+                        project_client=project_client,
+                        deployment_name="gpt-4o",
+                        instructions="You are a helpful assistant.",
+                        tools=["file_search"],
+                        metadata={"source": "AzureAIAgent"},
+                    )
+
+                    await agent_with_file_search.on_upload_for_file_search(
+                        file_paths=[
+                            "/workspaces/autogen/python/packages/autogen-core/docs/src/user-guide/core-user-guide/cookbook/data/product_info_1.md"
```

Comment:
We don't need to include a local file here. For this example, we can just download from a website e.g., `https://github.com/microsoft/autogen/README.md`. 

This way we can remove the data file.